src/headers.js

```diff
@@ -167,9 +167,9 @@ export default class Headers extends URLSearchParams {
 		return value;
 	}
 
-	forEach(callback) {
+	forEach(callback, thisArg) {
```

Comment:
```suggestion
	forEach(callback, thisArg = undefined) {
```

This allows `headers.forEach.length === 1`, which matches browsers.

Reply:
hmm a couple of things here:
1. I can add that for extra clarity but the end result feels the same as `thisArg` will be `undefined` when `headers.forEach.length === 1` anyways.
2. To match browsers, the default would be a global [object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#description) such as window. On node, I guess `global` would do the job. Try this: `[1].forEach(function () { console.log(this === global) })`

Reply:
> I can add that for extra clarity

The result indeed is the same inside `forEach`, but the `= undefined` is _essential_ to get the correct `.length` property. Try

```js
console.log(({ f(a, b) {} }).f.length);
console.log(({ f(a, b = undefined) {} }).f.length);
```

You will see that the first statement prints 2, while the second prints 1. We want the second behavior for `forEach`, and that's why you should add it. But indeed, the expected behavior is that if `thisArg` is not supplied, then it defaults to `undefined`.

By the way, the `.length` property of the `forEach` method is defined in the Web IDL standard, see ["define the iteration methods"](https://heycam.github.io/webidl/#define-the-iteration-methods) step 2.4.4.

> To match browsers, the default value would be a global object such as window.

That's not quite right. In browsers, the default value is `window` _only if_ we are not in [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode):

![image](https://user-images.githubusercontent.com/1538624/105444457-e03f4200-5c3b-11eb-8d74-e9d08938a162.png)

Our test environments are all strict mode, but you can run `Reflect.apply(function() { console.log(this); }, undefined, [])` in the Node.js REPL (which is sloppy by default) to see that `this` is actually `global` in Node.js too.

All this is to say, you don't have to do anything special to get the global object behavior. (In fact, you _cannot_ do anything here to replicate that in pure JavaScript, since there's no easy way of detecting whether a function is strict or sloppy.)

If you're interested in finding where this behavior is specified, it's in the ECMAScript standard, within [OrdinaryCallBindThis](https://tc39.es/ecma262/#sec-ordinarycallbindthis) abstract operation's steps 5–6.

Reply:
hey @TimothyGu ! Thanks for that! It was very informative. All suggestions are committed now. Let me know if there's anything else I should do.

Btw, there's just a lint issue going on that I'm not able to fix. :/